src/layers/normalization.h

```diff
@@ -172,4 +172,67 @@ class RMSNormImpl : public torch::nn::Module {
 };
 TORCH_MODULE(RMSNorm);
 
+// Root mean square normalization
+class RMSNormResidualImpl : public torch::nn::Module {
+ public:
+  RMSNormResidualImpl(int64_t dim,
+              float eps,
+              torch::ScalarType dtype,
+              const torch::Device& device)
+      : eps_(eps) {
+    weight_ = register_parameter(
+        "weight",
+        torch::empty({dim}, torch::dtype(dtype).device(device)),
+        /*requires_grad=*/false);
+  }
+
+  torch::Tensor forward(torch::Tensor input, torch::Tensor* residual) {
```

Comment:
Given that torch::Tensor functions somewhat like a shared pointer, adhering to the PyTorch convention, we can use torch::Tensor& for the out parameter. While this style may not be optimal and could obscure the purpose of in and out parameters, it aligns with PyTorch's conventions. Typically, pointers are preferred for out parameters, but for consistency, let's adhere to PyTorch's style for now.